query-parser/tests/find_tests.rs

```diff
@@ -208,3 +210,27 @@ fn can_parse_simple_or_and_join() {
                    )),
                ]);
 }
+
+#[test]
+fn can_parse_order_by() {
+    let invalid = "[:find ?x :where [?x :foo/baz ?y] :order]";
+    assert!(parse_find_string(invalid).is_err());
+
+    // Defaults to ascending.
+    let default = "[:find ?x :where [?x :foo/baz ?y] :order ?y]";
+    assert_eq!(parse_find_string(default).unwrap().order,
+               Some(vec![Order(Direction::Ascending, Variable::from_valid_name("?y"))]));
+
+    let ascending = "[:find ?x :where [?x :foo/baz ?y] :order (asc ?y)]";
```

Comment:
Because everything else in this formulation of Datalog is prefixed — aggregate calls are `(max ?x)`, inequalities are `(< ?x ?y)`, etc.

If we support ordering by expressions, we'll have things like `(asc (+ ?x ?y))` here.

I think you're suggesting `(?x :asc)`, which would extend to `((+ ?x ?y) asc)`, which is really unusual for a parenthesized language. Is that what you meant, or am I misunderstanding?